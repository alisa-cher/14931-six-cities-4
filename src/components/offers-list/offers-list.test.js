import React from 'react';
import renderer from 'react-test-renderer';
import OffersList from "../offers-list/offers-list.jsx";
import {BrowserRouter} from "react-router-dom";

it(`Offers list renders ok`, () => {
  const offers = [{
    type: `appartement`,
    price: 140,
    title: `just amazing1`,
    isPremium: true,
    rating: 4.3,
    previewPhoto: `img/apartment-01.jpg`,
  }];

  const tree = renderer
    .create(
        <BrowserRouter>
          <OffersList
            offers={offers}
            onCardTitleClick={() => {}}
            classNamePrefix={`just-some-prefix-for-tests`}
          />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
