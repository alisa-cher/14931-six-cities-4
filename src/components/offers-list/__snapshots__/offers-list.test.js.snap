// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers list renders ok 1`] = `
<div
  className="places__list just-some-prefix-for-tests__list"
>
  <article
    className="place-card just-some-prefix-for-tests__card"
    onMouseEnter={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="just-some-prefix-for-tests__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/property"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            140
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/property"
          onClick={[Function]}
        >
          just amazing1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        appartement
      </p>
    </div>
  </article>
</div>
`;
